### DISCUSSION TITLE
FleetDM is not showing osquery endpoints on fresh distributed manager node.

### DISCUSSION THREAD

**Discussion by Rbartram** [Security-Onion-Solutions/securityonion#6056] [Unsupported Versions] [answered]
Hello, I'm struggling to get FleetDM to show osquery endpoints.  I have it working on a test standalone instance of SO, but not in my distributed environment.  I even created a brand new distributed environment (only having a manager node at the moment) and my test endpoint still isn't showing up.  

I've uninstalled/reinstalled the kolide osquery package on my test endpoint using the proper msi from each environment.  I can switch back to the standalone environment and my endpoint checks in just fine.

- On the management node in each distributed environment, there are no logs in /opt/so/log/fleet, but logs do exist on the standalone setup.  
- so-allow has been run to allow osquery from my internal subnets (10.0.0.0/16).
- When I restart the LauncherSoLauncherSvc service on my endpoint, tcpdump port 8090 on my manager node show's packets from my endpoint hitting the manager node on the docker0 ip assigned to so-nginx.
- SecurityOnion 2.3.80 ISO was used to install both the standalone and test distributed manager node. 

I welcome a kick in the right direction here.

Thank you, ...Rob

**Comment by Rbartram** [comment]
I've finally been able to track the problem down to a x509 certificate issue.  When doing a distributed installation, I chose to use a different FQDN to access the SOC.  OSQuery is trying to connect using the FQDN, but the cert is for the local host name.

I found this by killing the "launcher" process on my distributed management node  (as it wasn't showing in Fleet either) and started it manually using the following:

`sudo /usr/local/so-launcher/bin/launcher -config /etc/so-launcher/launcher.flags`

This returned the following error (amongst other things)

{"caller":"request_enrollment.go:207","enrollSecret":"obscured","err":"rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"**transport: authentication handshake failed: x509: certificate is valid for som-01, not som.domain.com**\"","hostIdentifier":"781af472-cw5e-45c4-85d6-6eb2b550d684","method":"RequestEnrollment","nodekey":"","reauth":false,"severity":"info","took":"498.213µs","ts":"2021-11-01T14:40:08.710013587Z","uuid":"7d693236-1dcc-42be-9426-6ce96eb219b1"}

I'm going to search for how to resolve this, but I would appreciate any meaning advice in this regard.

Thank you, ...Rob

**Comment by Rbartram** [comment, marked as answer]
I seemed to have resolved this by doing the following: (condensed version of SuiNom's reply at https://github.com/Security-Onion-Solutions/securityonion/discussions/5274)

- Backed up /opt/so/saltstack/local/pillar/global.sls
- Edited  /opt/so/saltstack/local/pillar/global.sls and specified my custom FQDN for fleet_custom_hostname:
    `fleet_custom_hostname: 'som.domain.com'`
- Deleted /etc/pki/fleet.*  (these will get rebuilt when the manager server is restarted)
- Deleted /etc/pki/managerssl.*  (these will get rebuilt when the manager server is restarted)
- Rebooted my manager server

**Reply by Xboarder56** [reply]
Thanks for this solved my issue.
